src/main/java/org/zeromq/ZMonitor.java

```diff
@@ -44,7 +45,7 @@
         private ZEvent(ZMsg msg)
         {
             assert (msg != null);
-            assert (msg.size() == 3 || msg.size() == 4);
+            assert (msg.size() == 3 || msg.size() == 4) : msg.size();
```

Comment:
Similar to me other comment, it might be safer if we read the size only once:

```java
int size = msg.size();
assert (size == 3 || size == 4) : size;
```

Reply:
msg is passed into the constructor and is not used otherwise. There is no possibility it can be modified at that point in time.